interfaces/builtin/media_hub.go

```diff
@@ -0,0 +1,301 @@
+    interface=org.freedesktop.DBus
+    member="GetConnectionUnix{ProcessID,User}"
+    peer=(label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+	bus=session
+	name="org.mpris.MediaPlayer2.MediaHub",
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+	bus=session
+	name="core.ubuntu.media.Service",
+
+dbus (receive, send)
+	bus=session
+	path=/com/ubuntu/media/Service{,/**}
+	interface=org.freedesktop.DBus**
+	peer=(label=unconfined),
+
+dbus (send)
+	bus=session
+	path=/org/freedesktop/Telepathy/AccountManager
+	interface=org.freedesktop.DBus.Properties
+	member="GetAll",
+
+# We can always connect to ourselves
+dbus (receive)
+	bus=session
+	path=/core/ubuntu/media/Service
+	peer=(label=@{profile_name}),
+
+# Allow all access to powerd for now, but we can fine-tune this if needed
+dbus (receive, send)
+	bus=system
+	path=/com/canonical/powerd
+	interface=com.canonical.powerd,
+
+dbus (receive, send)
+	bus=system
+	path=/com/canonical/Unity/Screen
+	interface=com.canonical.Unity.Screen,
+`)
+
+var mediaHubConnectedSlotAppArmor = []byte(`
+# Allow connected clients to interact with the service
+
+# Allow connected clients to interact with the player
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Properties
+    path=/core/ubuntu/media/Service
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Introspectable
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (receive)
+    bus=session
+    interface="core.ubuntu.media.Service{,.*}"
+    path=/core/ubuntu/media/Service
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to manager Player sessions
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=core.ubuntu.media.Service
+    member="{Create,Detach,Reattach,Destroy,CreateFixed,Resume}Session"
+	peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to pause all other sessions
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=core.ubuntu.media.Service
+    member="PauseOtherSessions"
+	peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to query/modify service properties
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=org.freedesktop.DBus.Properties
+    member="{Get,Set}"
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (send)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=org.freedesktop.DBus.Properties
+    member=PropertiesChanged
+    peer=(label=###PLUG_SECURITY_TAGS###),
+`)
+
+var mediaHubConnectedSlotAppArmorClassic = []byte(`
+# Allow unconfined clients to interact with the player on classic
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    peer=(label=unconfined),
+
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Introspectable
+    peer=(label=unconfined),
+`)
```

Comment:
This should probably just be moved to mediaHubPermanentSlotAppArmor since it offers no security benefit on all snaps (unconfined is free to change apparmor profiles and is considered trusted).